### ISSUE TITLE
Invalid bidirectional type inference when assigning empty tuple

### ISSUE THREAD

**Issue by tkukushkin** [microsoft/pyright#3137]
**To Reproduce**
```python
x: tuple[()] = ()
reveal_type(x[0] + 1)  # information: Type of "x[0] + 1" is "Any"
```

**Expected behavior**
I expect an error, as it happens with mypy:
```
error: Tuple index out of range
```

**VS Code extension or command-line**
pyright 1.1.223

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.227, which I just published. It will also be included in the next release of pylance.
